include/geom/face_quad4.h

```diff
@@ -159,6 +159,27 @@ class Quad4 : public Quad
 
 };
 
+class QuadShell4 : public Quad4
```

Comment:
I think this new class should be in its own file.  Also it should have doxygen style documentation, e.g.
```
/**
 *
 */
```
at the top of the class.